### PR TITLE
Performance Automation Phase 1

### PR DESCRIPTION
**Pull request by navin22**
- Running performance projects using Benchmark PowerShell commandlet
- Running against nunit.console and xunit.console
- Comparing with Goal and asserting
[ToDo] : Will change the Goal numbers

### CODE REVIEW

**Comment by navin22** [comment]
Sample Output from the script.

`PS E:\repos\nvstest\scripts> .\perf.ps1 -f net452
Installing NuGet Package provider
Saving BenchMark module into tools folder
Setting InstallationPolicy to Trusted for PSGallery Repository
Installing Benchmark module
... Discovering Tests in E:\repos\nvstest\test\TestAssets\MSTestAdapterPerfTestProject\bin\Release\net452\MSTestAdapterPerfTestProject.dll using vstest.console
... Executing Tests in E:\repos\nvstest\test\TestAssets\MSTestAdapterPerfTestProject\bin\Release\net452\MSTestAdapterPerfTestProject.dll using vstest.console
... Discovering Tests in E:\repos\nvstest\test\TestAssets\NUnitAdapterPerfTestProject\bin\Release\net452\NUnitAdapterPerfTestProject.dll using vstest.console
... Executing Tests in E:\repos\nvstest\test\TestAssets\NUnitAdapterPerfTestProject\bin\Release\net452\NUnitAdapterPerfTestProject.dll using vstest.console
... Discovering Tests in E:\repos\nvstest\test\TestAssets\NUnitAdapterPerfTestProject\bin\Release\net452\NUnitAdapterPerfTestProject.dll using nunit.consolerunner
... Executing Tests in E:\repos\nvstest\test\TestAssets\NUnitAdapterPerfTestProject\bin\Release\net452\NUnitAdapterPerfTestProject.dll using nunit.consolerunner
... Discovering Tests in E:\repos\nvstest\test\TestAssets\XUnitAdapterPerfTestProject\bin\Release\net452\XUnitAdapterPerfTestProject.dll using vstest.console
... Executing Tests in E:\repos\nvstest\test\TestAssets\XUnitAdapterPerfTestProject\bin\Release\net452\XUnitAdapterPerfTestProject.dll using vstest.console
... Executing Tests in E:\repos\nvstest\test\TestAssets\XUnitAdapterPerfTestProject\bin\Release\net452\XUnitAdapterPerfTestProject.dll using xunit.runner.console


Machine Configuration


MachineName  : NAVB2.fareast.corp.microsoft.com
OSName       : Microsoft Windows 10 Enterprise
OSVersion    : 10.0.15063
MachineType  : x64-based PC
Processor    : Intel(R) Xeon(R) CPU E5-1650 0 @ 3.20GHz
LogicalCores : 12
RAMSize      : 15.925609588623




Runner              Adapter Action    ElapsedTime Goal    Delta Status PayLoad                          RunnerVersion AdapterVersion
------              ------- ------    ----------- ----    ----- ------ -------                          ------------- --------------
vstest.console      MsTest  Discovery      1906.1 3500   1593.9 PASS   MSTestAdapterPerfTestProject.dll 15.5.0-dev    1.2.0-beta
vstest.console      nUnit   Discovery     7254.67 6500  -754.67 FAIL   NUnitAdapterPerfTestProject.dll  15.5.0-dev    3.8.0
nunit.consolerunner         Discovery     2198.23 3500  1301.77 PASS   NUnitAdapterPerfTestProject.dll  3.7.0
vstest.console      xUnit   Discovery    15296.64 6500 -8796.64 FAIL   XUnitAdapterPerfTestProject.dll  15.5.0-dev    2.3.0-beta3-build3705



Runner               Adapter Action    ElapsedTime  Goal    Delta Status PayLoad                          RunnerVersion         AdapterVersion
------               ------- ------    -----------  ----    ----- ------ -------                          -------------         --------------
vstest.console       MsTest  Execution     2172.48  9500  7327.52 PASS   MSTestAdapterPerfTestProject.dll 15.5.0-dev            1.2.0-beta
vstest.console       nUnit   Execution     6818.52 45000 38181.48 PASS   NUnitAdapterPerfTestProject.dll  15.5.0-dev            3.8.0
nunit.consolerunner          Execution    35211.46 40000  4788.54 PASS   NUnitAdapterPerfTestProject.dll  3.7.0
vstest.console       xUnit   Execution    16272.98 50000 33727.02 PASS   XUnitAdapterPerfTestProject.dll  15.5.0-dev            2.3.0-beta3-build3705
xunit.runner.console         Execution    38847.45 38000  -847.45 FAIL   XUnitAdapterPerfTestProject.dll  2.3.0-beta3-build3705`

**Comment by navin22** [comment]
Sample output from the script:
image

File: scripts/perf.ps1 [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+    $osInfo.RAMSize=$sysinfo.TotalPhysicalMemory/1GB
+    $osInfo.MachineName += "."+$sysinfo.Domain
+    return New-Object -TypeName psobject -Property $osInfo
+}
+
+foreach ($src in $Script:TPT_SourceFolders)
+{
+    $assets = Get-ChildItem -Recurse -Path $src -Include *.csproj
+
+        foreach($payload in $Script:TPT_PayLoads)
```
Comment (codito) [inline scripts/perf.ps1]: nit: indent

**Review by codito — APPROVED** [review]

**Comment by singhsarab** [comment]
@dotnet-bot test Windows_NT / Release please

**Comment by singhsarab** [comment]
@dotnet-bot test Windows_NT / Release Build please

**Pull request merged**